### PR TITLE
Setup vscode tests

### PR DESCRIPTION
**Pull request by Draggu**
**Stack**:
- #32
- #31
- #30
- #29
- #22 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

**Review by mkaput — CHANGES_REQUESTED** [review]
I would like to see some actual tests of our extension before lgtming this

**Review by piotmag769 — APPROVED** [review]

File: eslint.config.mjs [hunk 1]
```diff
@@ -15,5 +16,10 @@ export default ts.config(
     languageOptions: {
       globals: globals.node,
     },
+    plugins: { "chai-friendly": pluginChaiFriendly },
+    rules: {
+      "@typescript-eslint/no-unused-expressions": "off", // disable original rule
+      "chai-friendly/no-unused-expressions": "error",
+    },
```
Comment (mkaput) [inline eslint.config.mjs:23]: use `pluginChaiFriendly.configs.recommendedFlat` instead of this

File: package.json [hunk 2]
```diff
@@ -173,7 +173,8 @@
     "lint": "npm run lint-fmt && npm run lint-eslint",
     "lint-eslint": "eslint .",
     "lint-fmt": "prettier --check .",
-    "test": "node --test"
+    "test": "cd test && node --test && cd ..",
```
Comment (mkaput) [inline package.json:176]:
```suggestion
    "test": "node --test **/test/**/*.{cjs,mjs,js}",
```

you're change breaks `npm test --watch`

File: tsconfig.json
Comment (mkaput) [inline tsconfig.json:1]: why are you changing this file?

**Review by mkaput — CHANGES_REQUESTED** [review]

File: eslint.config.mjs (hunk 1 above)
Comment (Draggu) [reply eslint.config.mjs:24]:
```
"@typescript-eslint/no-unused-expressions": "off"
```

must stay as this is a workaround. `pluginChaiFriendly.configs.recommendedFlat` disables `no-unused-expressions` not `@typescript-eslint/no-unused-expressions`

File: tsconfig.json
Comment (Draggu) [reply tsconfig.json:1]: Discussed offline

**Review by mkaput — APPROVED** [review]

**Pull request merged**